audio-matching/save_mfcc_to_db.py

```diff
@@ -0,0 +1,49 @@
+import utils.dbHelper as dbHelper
+import sys
+import os.path
+from numpy import load
+import numpy as np
+
+# USAGE python save_to_db.py [youtube url]
+vidUrl = sys.argv[1]
+
+# get youtube id
+vidId = vidUrl.replace('https://www.youtube.com/watch?v=', '')
+
+# Check if mfcc data for the video is saved locally
+mfccPath = "mfccs-amp-intervals/{:}.npy".format(vidId)
+
+if not os.path.isfile(mfccPath):
+  print('Mfcc data not found, make sure url is right and file exists')
+  exit()
+
+# load data
+# mfcc and amp data format: [start, end, amplitude, mfcc as string]
+print('loading mfcc, amp data')
+mfccData = load(mfccPath)
+
+rows = len(mfccData) // 4
+mfccData = mfccData.reshape(rows, 4)
+
+# convert mfccData to query syntax in the for of (url, start, mfcc, amp)
+def formatUpdatedValues(url, start, end, mfcc, amplitude):
+  return "('{:}', {:}, {:}, '{:}', {:})".format(url, start, end, mfcc, amplitude)
+
+# loop through mfccData, build array of update string
+updatedValues = []
+for (start, end, amplitude, mfcc_string) in mfccData:
+  valuesString = formatUpdatedValues(vidUrl, start, end, mfcc_string, amplitude)
+  updatedValues.append(valuesString)
+
+# build query sytax
+querySyntax = """ INSERT INTO labelled (url, start, end, mfcc, amplitude)
+                  VALUES
+                    {:}
+                  ON DUPLICATE KEY UPDATE
+                    mfcc = VALUES(mfcc),
+                    amplitude = VALUES(amplitude);""".format(",".join(updatedValues))
+
```

Comment:
Looks good, at this point I don't think it matters if we have more than one data entry query.

Reply:
Yeah this query is kinda confusing but its one of the only ways to update all rows in 1 query. How it works: If there is a duplicated key (aka url, start, end already exist in the db) it will update the mfcc, and amplitude of that row. Otherwise it will insert a new row (it shouldn't have to because all the rows should already be there)